ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/private/provision/cucushift-installer-rehearse-aws-ipi-private-provision-chain.yaml

```diff
@@ -0,0 +1,11 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-private-provision
+  steps:
+    - chain: ipi-conf-aws
```

Comment:
I guess we can use the flatten steps defined in ipi-conf-aws chain:
```
  - ref: ipi-conf
  - ref: ipi-conf-aws
  - ref: ipi-install-monitoringpvc
```
Once that, this make me remember https://github.com/openshift/release/pull/26774/files#diff-c3b68538742ec7d3c0722a1be46cf338174deb2d0be207ae40a7f7397a504143, we are trying to put `publish` setting in `ipi-conf`. Is it possible to merge the subnet setting into ` ipi-conf-aws`, once that, sounds like we can merge the ipi-conf-aws-privatecluster into different existing ref/chain.

Reply:
> I guess we can use the flatten steps defined in ipi-conf-aws chain:

Do you mean that the steps  in the `cucushift-installer-rehearse-aws-ipi-private-provision` will be more clear?

> Also Is it possible to merge the subnet setting into ipi-conf-aws?

`ipi-conf-aws` is responsible for common settings, and the settings relevant to a particular environment are placed in separate chains, e.g. `ipi-conf-aws-sharednetwork`, `ipi-conf-aws-blackholenetwork`, also the resource (VPC) were created inside them.
Some points that need to be considered:
1. compatibility with current chains/workflow
2. the map of subnets and AZs
3. separate the resource creation and conf setting (?) 

Personally, I prefer the current process - the sub-conf has full control over settings, and we will not break existing logic.

Reply:
Or we could deep dive into that in another PR.

Reply:
If we want to keep the same logic for the existing chains, we can keep this.

Per the existing ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-chain.yaml
```
  steps:
  - ref: ipi-conf
  - ref: ipi-conf-aws-sharednetwork
  - ref: ipi-conf-aws
  - ref: ipi-install-monitoringpvc
```
A minor question, can we modify this prviate cluster workflow like this:
```
  steps:
  - ref: ipi-conf
  - ref: aws-provision-vpc-shared
  - ref: aws-provision-bastionhost
  - ref: ipi-conf-aws-privatecluster
  - ref: ipi-conf-aws
  - ref: ipi-install-monitoringpvc
```

Reply:
OK, I will update.

Reply:
Once that, I am thinking if we can drop https://github.com/openshift/release/pull/22260/files#diff-0b7ac538a7b5359f9d723b381ff932e8a37b68b08d6f9028cbaea6cd410c6711R32-R37

Reply:
> Also Is it possible to merge the subnet setting into ipi-conf-aws?
Do you mean if we do this change? Yes, we can set AZ and subnet together.